### ISSUE TITLE
Trait ConvertBase64ToFiles does not work with nested data

### ISSUE THREAD

**Issue by andrzejkupczyk** [protonemedia/laravel-mixins#2]
Hey :wave: Unfortunately, current implementation does not work correctly. Let's assume that we have the following request:

```json
{
    "data": {
        "attributes": {
            "avatar": "YW55dGhpbmc="
        }
    }
}
```

Now, when you're using the Form Request in your controller, the base64 encoded data won't get properly validated. In addition, calling the file method on the $request instance will give you `null`.

The reason for this is that Laravel uses "dot" notation to access the arrays, while Symfony's ParameterBag treats dotted keys as they are. In result there is a file with the `data.attributes.avatar` key (so `$request->file('data.attributes.avatar')` returns `null`; you should probably use `$request->files->get('data.attributes.avatar')` instead), and at the same time validation will fail cause there is no file within the input attributes.

There is more. Removing parameter (https://github.com/protonemedia/laravel-mixins/blob/master/src/Request/ConvertsBase64ToFiles.php#L51) won't work either in this case. You should replace whole input for the request instead. For example:

```php
$this->replace(Arr::except($this->input(), $key));
```

On the other hand, if you'd like to support accessing file using "dot" notation and validation, it should probably look like:

```php
$attributes = $this->input();
Arr::set($attributes, $key, $uploadedFile);

$this->replace($attributes);
```

Then you could overwrite the "file" method in the ConvertsBase64ToFiles trait:

```php
public function file($key = null, $default = null)
{
    return $this->has($key) ? $this->input($key) : parent::file($key, $default);
}
```

**Comment by pascalbaljet** [comment]
Thanks for reporting! I'll look into this, but I'm not too keen on overwriting framework methods.

**Comment by andrzejkupczyk** [comment]
Neither do I, but you're already overwriting `prepareForValidation()` method anyway :slightly_smiling_face:

**Comment by pascalbaljet** [comment]
Fixed in v3.0.1:
https://github.com/protonemedia/laravel-mixins/releases/tag/3.0.1

**Comment by andrzejkupczyk** [comment]
It looks promising, thank you :1st_place_medal:
